File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/model/SinglePodcastModel.java
package com.pietrantuono.podcasts.singlepodcast.model;


import com.pietrantuono.podcasts.addpodcast.model.pojos.SinglePodcast;
import com.pietrantuono.podcasts.apis.PodcastFeed;

import rx.Observer;

public interface SinglePodcastModel {

    void subscribeToFeed(Observer<PodcastFeed> observer);

    void unsubscribe();

    void subscribeToIsSubscribedToPodcast(Observer<Boolean> observer);

    void startModel(SinglePodcast podcast);

    void onSubscribeUnsubscribeToPodcastClicked();
}

File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/model/SinglePodcastModel.kt
package com.pietrantuono.podcasts.addpodcast.singlepodcast.model


import com.pietrantuono.podcasts.apis.PodcastFeed
import models.pojos.Podcast

import rx.Observer

interface SinglePodcastModel {

    fun subscribeToFeed(observer: Observer<PodcastFeed>, timeOutTime: Long)

    fun unsubscribe()

    fun subscribeToIsSubscribedToPodcast(observer: Observer<Boolean>)

    fun startModel(podcast: Podcast?)

    fun onSubscribeUnsubscribeToPodcastClicked()

    fun saveFeed(podcastFeed: PodcastFeed?)

    var podcastFeed: PodcastFeed?

    val hasEpisodes: Boolean

    var alreadyAttemptedToGetFeed: Boolean
}
